mlflow/server/__init__.py

```diff
@@ -191,10 +205,15 @@ def _run_server(
     if expose_prometheus:
         env_map[PROMETHEUS_EXPORTER_ENV_VAR] = expose_prometheus
 
-    app_spec = f"{__name__}:app" if app_name is None else _find_app(app_name)
+    if app_name is None:
+        app = f"{__name__}:app"
+    else:
+        app = _find_app(app_name)
+        app = f"{app}()" if _is_factory(app) else app
```

Comment:
i think we don't need the () for waitress

https://flask.palletsprojects.com/en/2.3.x/deploying/waitress/#running

Reply:
i got error when running a plugin app

```
1. Malformed application 'my_auth:create_app()'
```

Reply:
On windows?

Reply:
yes

Reply:
Ok, let's see what happens if we remove `()` on windows. Pushing a commit.

Reply:
i tried adding `--call` as opt and removing `()`, worked on my windows

Reply:
Done.

Reply:
I see, we need `--call`.

Reply:
Pushed another commit.

Reply:
@gabrielfu Did it work?

Reply:
yes it's working now after putting `--call` before app name!